sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -986,10 +986,13 @@ private async Task<AmqpSession> CreateAndOpenSessionAsync(AmqpConnection connect
             {
                 var openObjectCompletionSource = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
 
-                using var registration = cancellationToken.Register(static state =>
+                using var registration = cancellationToken.Register(state =>
                 {
                     var tcs = (TaskCompletionSource<object>)state;
-                    tcs.TrySetCanceled();
+                    if (tcs.TrySetCanceled())
+                    {
+                        target.SafeClose();
```

Comment:
Isn't SafeClose itself another operation that has a default timeout associated?

```
this.BeginClose(AmqpConstants.DefaultTimeout, AmqpObject.onSafeCloseComplete, (object) this);
```

that's why I used this approach https://github.com/Azure/azure-sdk-for-net/pull/19888/files#diff-95ce46ce338976bfb451cfcd8b5da1d77b2c4b2d9e79de0b3d04812be897e7ffR297

Reply:
We probably want to consider using Daniel's approach or possibly calling [`Abort`](https://github.com/Azure/azure-amqp/blob/f9d7c10a6c77dfcc0189829c1f6a51abd8c45592/src/AmqpObject.cs#L359) here.   We may also want to follow the logic in the `catch` block to address the session (also using a raw `Close` or `Abort`).

That said, I don't think the AMQP library behavior is well documented here nor well understood.   We should probably seek guidance from the AMQP library maintainers for this;  an internal email thread is probably more likely to catch their attention than a PR mention.

Reply:
This recommendation was from the AMQP library maintainer. Yes, there is a timeout to this operation but it is not a long polling operation (going along with our general understanding of when we would use cancellation). Also in the catch for this method, we call SafeClose on the sessions.

Reply:
I don't see any mention of a specific method on the email thread where this was discussed.   Did I miss a fork where `SafeClose` was explicitly recommended?    I also found the discussion in that thread to be at a very high level, and am not sure that the corner cases were all accounted for.

Reply:
This was discussed over Teams - the advice was to close the link. When not using SafeClose, exceptions could occur if the link was never actually opened.